### PR TITLE
SC-2566: ".ssh/known_hosts" should be updated with ssh github.com command

### PR DESCRIPTION
**Pull request by zyuzka**
#### Overview

- Ticket: https://spryker.atlassian.net/browse/SC-2566
- RG: https://release.spryker.com/release-groups/view/2401

###### Change log

Implemented functionality for a generation `.ssh/known_hosts` into containers on `build` step

### CODE REVIEW

File: generator/src/templates/deploy.bash.twig [hunk 1]
```diff
@@ -61,6 +61,10 @@ export SPRYKER_STORE=${SPRYKER_STORE:-${APPLICATION_STORE:-${SPRYKER_DEFAULT_STO
 readonly SPRYKER_SELF_SCRIPT=${SPRYKER_SELF_SCRIPT:-$(basename $0)}
 readonly GITHUB_TOKEN=${GITHUB_TOKEN:-''}
 
+{% if _knownHosts is defined %}
+export KNOWN_HOSTS="{{ _knownHosts }}"
+{% endif %}
```
Comment (sprymiker) [inline generator/src/templates/deploy.bash.twig]: Please remove if. Let's always have the variable
`export KNOWN_HOSTS="{{ _knownHosts | default("") }}"`

File: generator/index.php [hunk 2]
```diff
@@ -19,6 +19,13 @@
 
 $projectData = $yamlParser->parseFile($projectYaml);
 
+$knownHostsYaml = $deploymentDir . DS . 'known-hosts.yml';
```
Comment (sprymiker) [inline generator/index.php]: $projectData['_knownHosts'] = buildKnownHosts($projectData...)

Comment (zyuzka) [reply generator/index.php]: Fixed

File: generator/src/templates/deploy.bash.twig (hunk 1 above)
Comment (zyuzka) [reply generator/src/templates/deploy.bash.twig]: fixed

File: generator/index.php [hunk 3]
```diff
@@ -360,3 +362,99 @@ function buildEndpointMapByStore(array $projectGroups): array
...
 }
+
+/**
+ * @param string $deploymentDir
+ *
+ * @return string
+ */
+function buildKnownHosts(string $deploymentDir): string
+{
+    $knownHostsYamlPath = $deploymentDir . DS . '.known_hosts';
```
Comment (sprymiker) [inline generator/index.php]: yaml?

File: generator/index.php [hunk 4]
```diff
@@ -360,3 +362,99 @@ function buildEndpointMapByStore(array $projectGroups): array
...
+}
+
+/**
+ * @param string $knownHost
+ *
+ * @return bool
+ */
+function isHostValid(string $knownHost): bool
+{
+    if (!isIp($knownHost) && !isHost($knownHost)) {
```
Comment (sprymiker) [inline generator/index.php]: return isIp($knownHost) || isHost($knownHost);

File: generator/index.php [hunk 5]
```diff
@@ -360,3 +362,99 @@ function buildEndpointMapByStore(array $projectGroups): array
...
+ */
+function getKnownHosts(string $knownHostsYamlPath): array
+{
+    $knownHosts = file_get_contents($knownHostsYamlPath);
+
+    if (!$knownHosts) {
+        return [];
+    }
+
+    $knownHosts = array_filter(preg_split('/[\s]+/', $knownHosts));
```
Comment (sprymiker) [inline generator/index.php]: array_filter(..., function ($host) { return $knownHost 
&& isHostValid($knownHost)
};

File: generator/index.php (hunk 4 above)
Comment (zyuzka) [reply generator/index.php]: Fixed

File: generator/index.php (hunk 5 above)
Comment (zyuzka) [reply generator/index.php]: Fixed

**Pull request merged**